db-tests/src/test/java/gov/nasa/jpl/aerie/database/ExternalEventTests.java

```diff
@@ -0,0 +1,1627 @@
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertThrowsExactly;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+@SuppressWarnings("SqlSourceToSinkFlow")
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class ExternalEventTests {
+
+  /*
+    ---------- GENERIC DATABASE TESTING SETUP ----------
+   */
+  private DatabaseTestHelper helper;
+  private MerlinDatabaseTestHelper merlinHelper;
+  private Connection connection;
+
+  @AfterEach
+  void afterEach() throws SQLException {
+    helper.clearSchema("merlin");
+  }
+
+  @BeforeAll
+  void beforeAll() throws SQLException, IOException, InterruptedException {
+    helper = new DatabaseTestHelper("external_event", "Activity Directive Changelog Tests");
+    connection = helper.connection();
+    merlinHelper = new MerlinDatabaseTestHelper(connection);
+  }
+
+  @AfterAll
+  void afterAll() throws SQLException, IOException, InterruptedException {
+    helper.close();
+    connection = null;
+    helper = null;
+  }
+
+
+  /*
+    ---------- COMMONLY REPEATED VARIABLES ----------
+   */
+  // Source Type (st)
+  final String st = "Test";
+
+  // Derivation Group (dg)
+  final String dg = "Test Default";
+
+  // Event Type (et)
+  final String et = "Test";
+
+  // Metadata/Properties (mt)
+  final String mt = "{}";
+
+  // Created At (ca)
+  final String ca = "2024-01-01T00:00:00Z";
+
+
+  /*
+    ---------- COMMONLY REPEATED FUNCTIONS ----------
+   */
+  /**
+   * A simple function to compare a SQL result (res) for a given key (key) against a list of expected entries (expected)
+   */
+  void compareLists(String[] expected, ResultSet res, String key) throws SQLException {
+    ArrayList<String> actual = new ArrayList<>();
+    while (res.next()) {
+      actual.add(res.getString(key));
+    }
+
+    assertEquals(expected.length, actual.size());
+    for (int i = 0; i < expected.length; i++) {
+      assertEquals(expected[i], actual.get(i));
+    }
+  }
+
+  /**
+   * Returns a java.sql.Statement object, from which queries can be run. The exact idiom, including the assertion,
+   *    are repeated fairly often, so it was extracted into a method for ease of readability and consistency.
+   * As we need to declare it outside of the assertion and assign it _in_ the assertion (else it cannot be returned)
+   *    we must wrap it as an AtomicReference object. It helps to isolate and check this assertion separately as if
+   *    _this_ fails we know our error is not with the underlying schema, but rather the connection.
+   */
+  AtomicReference<Statement> getStatement() {
+    final AtomicReference<Statement> ret = new AtomicReference<>();
+    assertDoesNotThrow(() -> ret.set(connection.createStatement()));
+    return ret;
+  }
```

Comment:
Please remove this method and instead follow the Aerie code style of using a `try-with-resources` to create a Statement when you want to execute a SQL statement.

```java 
 try(final var statement = connection.createStatement())